sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -419,66 +513,98 @@ struct tpu_params<tpu::amx, Ta, Tb, Tc, 0, 0, 0,
-  struct combination {
-    uint32_t max_msize;
-    uint32_t max_nsize;
-    uint32_t max_ksize;
-    uint32_t msize;
-    uint32_t nsize;
-    uint32_t ksize;
-    matrix_type atype;
-    matrix_type btype;
-    matrix_type accumulatortype;
-  };
-  
-  static constexpr combination combinations[] = {
-      {16, 16, 64, 0, 0, 0, matrix_type::sint8, matrix_type::sint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::sint8, matrix_type::uint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::uint8, matrix_type::sint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::uint8, matrix_type::uint8, matrix_type::sint32},
-      {16, 16, 32, 0, 0,0, matrix_type::bf16, matrix_type::bf16, matrix_type::fp32}};
-  static constexpr int num_combinations =
-      sizeof(combinations) / sizeof(combination);
+  using joint_matrix_accumulator = joint_matrix<Group, Tc,
+  use::accumulator, M, N>;
 };
-
-
 enum class tpu {
   xmx8,
   xmx16,
   amx
 };
+```
+===== Validation Example:
+```c++
+// User can provide sizes besides the types and tpu_params can assert
+  if they are supported or not
+// in this case, an assertion will happens as 16 is not a supported size for M
+using myparams = tpu_params<tpu::xmx16, int8_t, int8_t, int, 16, 16, 32>;
+size_t NDRangeM = M / myparams::M;  //Assertion would happen at this line
+size_t NDRangeN = N / myparams::N;
+```
 
+===== Default Values Example:
+```c++
+using myparams = tpu_params_both<tpu::xmx16, int8_t, int8_t, int>;
+// use this to construct the ranges on the host side
+size_t NDRangeM = M / myparams::M;
+size_t NDRangeN = N / myparams::N;
+//if M, N, K do not multiply the default sizes, padding has to be done
+// device code: the matrices are constructed using the default dimensions
+myparams::joint_matrix_a<sub_group, layout::row_major> sub_a;
+myparams::joint_matrix_b<sub_group, layout::row_major> sub_b;
+myparams::joint_matrix_accumulator<sub_group> sub_c;
+
+```
+==== Runtime Query
+This provides a more general query interface with information about
+sizes, types,  and scopes that are supported by a specific matrix
+implementation. This is needed to avoid padding by the user, for
+tuning, and efficient code generation if used by a library. The
+general query returns an array of `combinations` of `combination`
+type. Each combination includes the sizes and the types for the
+matrices A, B, and accumulator. Note that for each matrix hardware,
+the query returns `max_msize, max_nsize, max_ksize` or `msize, nsize,
+ksize` exclusively, depending on whether the implementation supports a
+continuous or discrete number of sizes. For example, the Intel AMX
+implementation supports a continuous number of sizes, so the `max_*`
+variant is applied and only the maximum number is returned. The Intel
+XMX implementation, on the other hand, supports a discrete list of
+numbers so the `msize, nsize, ksize` variant is applied.
```

Comment:
Reword this so that it does not talk about specific Intel hardware.  Instead, talk in general terms about some devices supporting discrete sizes while others support a range.